src/menu.ts

```diff
@@ -22,15 +22,94 @@ export class Menu {
+                },
+                    {
+                        label: "Start model Qwen2.5-Coder-3B-Q8_0-GGUF (<= 16GB VRAM)",
+                        description: `Requires brew, installs/upgrades llama.cpp server, downloads the model if not available, and runs llama.cpp server`
+                    },
+                    {
+                        label: "Start model Qwen2.5-Coder-7B-Q8_0-GGUF (> 16GB VRAM)",
+                        description: `Requires brew, installs/upgrades llama.cpp server, downloads the model if not available, and runs llama.cpp server`
+                    },
+                    {
+                        label: "Start model Qwen2.5-Coder-1.5B-Q8_0-GGUF (CPU Only)",
+                        description: `Requires brew, installs/upgrades llama.cpp server, downloads the model if not available, and runs llama.cpp server`
+                    }, 
+                    {
+                        label: "Start model Qwen2.5-Coder-0.5B-Q8_0-GGUF (CPU Only)",
+                        description: `Requires brew, installs/upgrades llama.cpp server, downloads the model if not available, and runs llama.cpp server`
+                    })
+            }
+
+            menuItems.push({
+                    label: "Start llama.cpp server with custom command from launch_cmd property",
+                    description: `Runs the command from property launch_cmd`
+                },
+                {
+                    label: "Stop llama.cpp server",
+                    description: `Stops llama.cpp server if it was started from llama.vscode menu."`
+                })
+
+            if (process.platform === 'darwin') { // if mac os
+                menuItems.push({
+                    label: "Uninstall llama.cpp server",
+                    description: `Requires brew, runs "brew uninstall llama.cpp"`
+                })
             }
-        ].filter(Boolean) as vscode.QuickPickItem[];
+            
+            
+
+        return menuItems.filter(Boolean) as vscode.QuickPickItem[];
     }
 
     handleMenuSelection = async (selected: vscode.QuickPickItem, currentLanguage: string | undefined, languageSettings: Record<string, boolean>) => {
+        const DEFAULT_PPORT_FIM_MODEL = "8012"
+        let endpointParts = this.app.extConfig.endpoint.split(":");
+        let port = endpointParts[endpointParts.length -1]
+        if (!Number.isInteger(Number(port))) port =  DEFAULT_PPORT_FIM_MODEL
+        let llmMacVramModelTemplate = " brew install llama.cpp && brew upgrade llama.cpp && llama-server -hf [model] --port " + port + " -ngl 99 -fa -ub 1024 -b 1024 --ctx-size 0 --cache-reuse 256"
+        let llmMacCpuTemplate = " brew install llama.cpp && brew upgrade llama.cpp && llama-server -hf [model] --port " + port + " -c 2048 -ub 1024 -b 1024 -dt 0.1 --ctx-size 0 --cache-reuse 256"
```

Comment:
- I think `brew install` automatically performs an upgrade so no need to add `brew upgrade`
- `-c 2048` is not used in the CPU template because `--ctx-size 0` will override it. Simply remove `-c 2048`
- You can simplify these commands using the new preset arguments available in the latest `llama.cpp` version:

![image](https://github.com/user-attachments/assets/84e96387-9307-4f8f-a395-a9acfeddba90)